docs/book/src/tasks/experimental-features/cluster-classes.md

```diff
@@ -0,0 +1,497 @@
+# Cluster Class
+
+The ClusterClass feature introduces a new way to create clusters which reduces boilerplate required and allows flexible and powerful customization of CAPI-created clusters.
+ClusterClass is implemented on top of pre-existing CAPI interfaces and will 
+
+This tutorial covers using Cluster API and Cluster Class to create a Kubernetes cluster and how to perform a one-touch Kubernetes update procedure using ClusterClass features.
+
+<aside class="note warning">
+
+<h1>Note</h1>
+
+This guide covers creating a ClusterClass cluster with the CAPD provider.
+</aside>
+
+## Installation
+
+### Common Prerequisites
+This guide is based on having the same pre-requisites as the CAPI [quick start guide].
+That is:
+- Install and setup [kubectl]  and [clusterctl]
+- Install [Kind] and [Docker]
+
+ClusterClass is currently behind a feature gate that needs to be enabled. To do so run:
+
+```bash
+export  EXP_CLUSTER_RESOURCE_SET=true
+export CLUSTER_TOPOLOGY=true
+```
+This ensures that Cluster Topology and Cluster Resource Set are enabled when we initialize our CAPI Providers.
+*Note*: Cluster Resource Set is not required for ClusterClass but is used in this tutorial to set up networking.
+
+A script to set up a Kind cluster pre-configured for CAPD can be found in the hack folder. To set up the cluster run:
+```bash
+./hack/kind-install-for-capd.sh
+clusterctl init --infrastructure docker
+````
+
+Alternatively the quickstart guide for setting up a CAPD enabled cluster can be followed.
+
+#### Creating the ClusterClass and templates
+Now that we have a CAPI Management Cluster with CAPD and Cluster Topology enabled we can create the ClusterClass and the templates needed to stamp clusters.
+
+The below contains six yaml objects which are used to define the cluster:
+
+1) ClusterClass
+1) DockerMachineTemplate
+1) DockerClusterTemplate
+1) DockerMachineTemplate for ControlPlane
+1) KubeadmControlPlaneTemplate
+1) KubeadmConfigTemplate
+
+The ClusterClass - first in the below yaml - contains references to the templates needed to build a full cluster using the shape defined in the ClusterClass.
+```yaml
+apiVersion: cluster.x-k8s.io/v1alpha4
+kind: ClusterClass
+metadata:
+  name: cluster-class
+  namespace: default
+spec:
+  controlPlane:
+    metadata:
+    ref:
+      apiVersion: controlplane.cluster.x-k8s.io/v1alpha4
+      kind: KubeadmControlPlaneTemplate
+      name: my-cluster-control-plane
```

Comment:
I wonder if we should call the ClusterClass `my-cluster-class` and then all the referenced templates `my-cluster-class-<suffix>`. The current names implicate a bit that they are for my-cluster and not for my-cluster-class